File: Frontend/AppFavoresMovil/app/src/main/java/com/example/appfavoresmovil/data/modelos/autenticacion/UsuarioPublico.kt
package com.example.appfavoresmovil.data.modelos.autenticacion

import com.google.gson.annotations.SerializedName

data class UsuarioPublico(
    @SerializedName("_id") val id: String,
    @SerializedName("correo") val correo: String,
    @SerializedName("nombreUsuario") val nombreUsuario: String,
    @SerializedName("descripcion") val descripcion: String,
    @SerializedName("fotoPerfil") val fotoPerfil: String,
    @SerializedName("calificacionMedia") val calificacionMedia: Float?,
    @SerializedName("rol") val rol: String
)

File: Frontend/AppFavoresEscritorio/AppFavoresEscritorio/Modelos/Autenticacion/UsuarioPublico.cs
using System.Text.Json.Serialization;

namespace AppFavoresEscritorio.Modelos.Autenticacion;

public class UsuarioPublico
{
    [JsonPropertyName("_id")]
    public string Id { get; set; } = string.Empty;

    [JsonPropertyName("correo")]
    public string Correo { get; set; } = string.Empty;

    [JsonPropertyName("nombreUsuario")]
    public string NombreUsuario { get; set; } = string.Empty;

    [JsonPropertyName("descripcion")]
    public string Descripcion { get; set; } = string.Empty;

    [JsonPropertyName("fotoPerfil")]
    public string FotoPerfil { get; set; } = string.Empty;

    [JsonPropertyName("calificacionMedia")]
    public float? CalificacionMedia { get; set; }

    [JsonPropertyName("rol")]
    public string Rol { get; set; } = "Usuario";
}
